### ISSUE TITLE
[Bug] [build] - Build on OSX fails due to symbol `at_quick_exit` not found

### ISSUE THREAD

**Issue by sskeirik** [runtimeverification/k#1580]
On commit 3d804961c, doing `mvn package -DskipTests` on OSX fails because the LLVM backend build step fails with this message:

```
[exec] /Users/skeirik/work/kframework/master/llvm-backend/src/main/native/llvm-backend/runtime/meta/ffi.cpp:276:47: error: use of undeclared identifier 'at_quick_exit'
[exec]     privateSymbols["at_quick_exit"] = (void *)at_quick_exit;
[exec]                                               ^
[exec] 1 error generated.
```

Using `make VERBOSE=1`, the full command that was run by maven/CMake/Make which repros the issue on my OSX box:

```
cd /Users/skeirik/work/kframework/master/llvm-backend/target/build/runtime/meta &&
/usr/local/opt/llvm/bin/clang++
  -I/usr/local/opt/llvm/include
  -I/Users/skeirik/work/kframework/master/llvm-backend/src/main/native/llvm-backend/include
  -I/Users/skeirik/work/kframework/master/llvm-backend/target/build/include
  -isystem /usr/local/include
  -isystem /usr/local/Cellar/libffi/3.3/include
  -fno-stack-protector
  -g
  -O3
  -DNDEBUG
  -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk
  -Wall
  -Werror
  -Wno-return-type-c-linkage
  -std=c++17
  -o CMakeFiles/meta.dir/ffi.cpp.o
  -c /Users/skeirik/work/kframework/master/llvm-backend/src/main/native/llvm-backend/runtime/meta/ffi.cpp
```

Googling around, it seems that `at_quick_exit` and `quick_exit` aren't available on OSX. Some references:
- [StackOverflow Post 1](https://stackoverflow.com/questions/30408197/how-to-compile-with-c11-standard-library-on-os-x-with-clang)
- [StackOverflow Post 2](https://stackoverflow.com/questions/11681403/state-of-c11-standard-support-in-libc)

[Boost 1.69.0](https://www.boost.org/users/history/version_1_69_0.html) provides `quick_exit` but it's not clear if it also provides `at_quick_exit`.

EDIT: CI builds are failing for the same reason.

**Comment by sskeirik** [comment]
Closed by #1581

**Comment by askunix** [comment]
A frustrating side note on std::quick_exit() and std::at_quick_exit() functions. They are still unimplemented in macOS's libc even after several years. Also there is a possible vulnerability in the C++ standard, where it states you can safely call std::quick_exit() from a signal handler, but it doesn't state the functions that are registered by std::at_quick_exit() must also meet the same requirements a regular signal handler does. I believe that might be the reason behind why those functions are not implemented yet.
